x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -348,60 +347,60 @@ export const updateCurrentWriteIndices = async (
 ): Promise<void> => {
   if (!templates.length) return;
 
-  const allIndices = await queryIndicesFromTemplates(callCluster, templates);
+  const allIndices = await queryDataStreamsFromTemplates(callCluster, templates);
   if (!allIndices.length) return;
-  return updateAllIndices(allIndices, callCluster);
+  return updateAllDataStreams(allIndices, callCluster);
 };
 
-function isCurrentIndex(item: CurrentIndex[] | undefined): item is CurrentIndex[] {
+function isCurrentDataStream(item: CurrentDataStream[] | undefined): item is CurrentDataStream[] {
   return item !== undefined;
 }
 
-const queryIndicesFromTemplates = async (
+const queryDataStreamsFromTemplates = async (
   callCluster: CallESAsCurrentUser,
   templates: TemplateRef[]
-): Promise<CurrentIndex[]> => {
-  const indexPromises = templates.map((template) => {
-    return getIndices(callCluster, template);
+): Promise<CurrentDataStream[]> => {
+  const dataStreamPromises = templates.map((template) => {
+    return getDataStreams(callCluster, template);
   });
-  const indexObjects = await Promise.all(indexPromises);
-  return indexObjects.filter(isCurrentIndex).flat();
+  const dataStreamObjects = await Promise.all(dataStreamPromises);
+  return dataStreamObjects.filter(isCurrentDataStream).flat();
 };
 
-const getIndices = async (
+const getDataStreams = async (
   callCluster: CallESAsCurrentUser,
   template: TemplateRef
-): Promise<CurrentIndex[] | undefined> => {
+): Promise<CurrentDataStream[] | undefined> => {
   const { templateName, indexTemplate } = template;
-  // Until ES provides a way to update mappings of a data stream
-  // get the last index of the data stream, which is the current write index
   const res = await callCluster('transport.request', {
     method: 'GET',
     path: `/_data_stream/${templateName}-*`,
   });
   const dataStreams = res.data_streams;
   if (!dataStreams.length) return;
   return dataStreams.map((dataStream: any) => ({
-    indexName: dataStream.indices[dataStream.indices.length - 1].index_name,
+    dataStreamName: dataStream.name,
     indexTemplate,
   }));
 };
 
-const updateAllIndices = async (
-  indexNameWithTemplates: CurrentIndex[],
+const updateAllDataStreams = async (
+  indexNameWithTemplates: CurrentDataStream[],
   callCluster: CallESAsCurrentUser
 ): Promise<void> => {
-  const updateIndexPromises = indexNameWithTemplates.map(({ indexName, indexTemplate }) => {
-    return updateExistingIndex({ indexName, callCluster, indexTemplate });
-  });
-  await Promise.all(updateIndexPromises);
+  const updatedataStreamPromises = indexNameWithTemplates.map(
```

Comment:
This should be moved inside the `Promise.all`. If `updateExistingDataStream` throws/rejects before the `await Promise.all` line we'll crash due to `UnhandledPromiseRejection`

Reply:
@jfsiii Can you elaborate or provide revised code? `updatedataStreamPromises` is passed to the `Promise.all` and `updateExistingDataStream` is async?

Also, I didn't change the logic here, just renaming. Due to being this a priority fix, I might merge this and follow up on code improvements based on your feedback afterwards.